### ISSUE TITLE
Error while adding host in Dashboard

### ISSUE THREAD

**Issue by lals1** [OpenAttestation/OpenAttestation#29]
I get below error while adding host in dashboard.

Trust Agent failed with code '1007' and message 'Trust Agent: Error while creating identity.' [AS_TRUST_AGENT_ERROR] [1012].

I read some where that I 

am missing EC.cer in /opt/intel/cloudsecurity/trustagent directory so I run command service tagent fixek to generate it but i get below error (attached is complate log file)

javax.xml.ws.soap.SOAPFaultException: java.security.KeyStoreException: Uninitialized keystore

I am using RHEL 7.1 and instruction followed from the link:
https://github.com/OpenAttestation/OpenAttestation/wiki/OAT-v2.2-RHEL7

P.S I am running OAT server and client on same machine.

tagent log: http://pastebin.com/myTH80WT

tomcat log: http://pastebin.com/FhGxTH8N

**Comment by lals1** [comment]
Any Help?

**Comment by ZahediAquino** [comment]
Were you able to install the trust agent on the host without any problem?

On the host, try running 'tagent fixek', if that doesn't work, uninstall trust agent 'tagent uninstall', reboot, in the bios, clear tpm ownership, reboot, enable tpm again, reboot, reinstall trust agent and try to add the host again.

Was this also setup using the deploy scripts with default settings? or did you customize something?

**Comment by lals1** [comment]
Yes, trust agent installation went error free. I have tried all the steps you mentioned above but no success.

Yes, the setup is deployed using script with default setting. The funny thing is that i have added one host running RHEL 7.1 in dashboard without any problem. It has exactly the same setting as the one which gets problems. I have tried another adding another host running Ubuntu but it also gets the same error.

The only thing I am doubting is trousers service log: have a look at it.

root@test-Precision-M4400t# service tcsd status
● trousers.service - LSB: starts tcsd
   Loaded: loaded (/etc/init.d/trousers)
   Active: active (running) since Wed 2015-10-21 10:51:28 EDT; 11min ago
     Docs: man:systemd-sysv-generator(8)
   CGroup: /system.slice/trousers.service
           └─1481 /usr/sbin/tcsd

Oct 21 10:51:28 test-Precision-M4400.localdomain systemd[1]: Starting LSB: starts tcsd...
Oct 21 10:51:28 test-Precision-M4400.localdomain trousers[1436]: * Starting Trusted Computing daemon tcsd
Oct 21 10:51:28 test-Precision-M4400.localdomain tcsd[1450]: TCSD TDDL[1450]: TrouSerS ioctl: (25) Inappropriate ioctl for device
Oct 21 10:51:28 test-Precision-M4400.localdomain tcsd[1450]: TCSD TDDL[1450]: TrouSerS Falling back to Read/Write device support.
Oct 21 10:51:28 test-Precision-M4400.localdomain TCSD[1481]: TrouSerS trousers 0.3.13: TCSD up and running.
Oct 21 10:51:28 test-Precision-M4400.localdomain trousers[1436]: ...done.
Oct 21 10:51:28 test-Precision-M4400.localdomain systemd[1]: Started LSB: starts tcsd.
Oct 21 10:54:18 test-Precision-M4400.localdomain tcsd[1481]: TCSD TCS[1481]: TrouSerS Unloading a public key of size 0!
root@test-Precision-M4400#

**Comment by ZahediAquino** [comment]
Ok, there are a couple more things we can check:

1. Can you please specify what java is actually being used by the host and server (java -version)?
2. Check your java.security file, make sure this is listed:
-security.provider.10=org.bouncycastle.jce.provider.BouncyCastleProvider
-Make sure this line is commented out if you have it:
jdk.tls.disabledAlgorithms=SSLv3, DH keySize < 768

**Comment by lals1** [comment]
Hi  yes the java is same on both client and server. here is the details:

java version "1.7.0_79"
OpenJDK Runtime Environment (IcedTea 2.5.6) (7u79-2.5.6-0ubuntu1.14.04.1)
OpenJDK 64-Bit Server VM (build 24.79-b02, mixed mode)

yes, the security.provider.10=org.bouncycastle.jce.provider.BouncyCastleProvider is present in java.security on both sides and jdk.tls.disabledAlgorithms=SSLv3, DH keySize < 768 is commented out as well. Also bcprov-jdk15-1.46.jar is copied to the respective directory on both sides.

I have tried disabling selinux and firewall but still same problem.
